test/Main.hs

```diff
@@ -686,16 +689,39 @@ streamOperations (stream, list, len) = do
         it "last empty" $ A.last stream `shouldReturn` Nothing
         it "maximum empty" $ A.maximum stream `shouldReturn` Nothing
         it "minimum empty" $ A.minimum stream `shouldReturn` Nothing
+        it "null empty" $ A.null stream `shouldReturn` True
     else do
         it "head nonEmpty" $ A.head stream `shouldReturn` Just (head list)
         it "last nonEmpty" $ A.last stream `shouldReturn` Just (last list)
         it "maximum nonEmpty" $ A.maximum stream `shouldReturn` Just (maximum list)
         it "minimum nonEmpty" $ A.minimum stream `shouldReturn` Just (minimum list)
+        it "null nonEmpty" $ A.null stream `shouldReturn` False
 
     where
+    expectedTail [] = Nothing
+    expectedTail (_:xs) = Just xs
 
     -- XXX run on empty stream as well
     transform streamOp listOp =
         (A.toList $ streamOp stream) `shouldReturn` listOp list
 
+    transformMaybe streamOp listOp = do
```

Comment:
How about something like this?
```
case list of
    _ : _ : _ -> it "tail" $ A.tail stream >>= toListSerial . fromJust `shouldReturn` tail list 
    _ -> it "tail empty or singleton" $ A.tail stream `shouldReturn` Nothing
```

Reply:
Could you that but I wanted something, which I could reuse for `init`

Reply:
We can use a similar two line code for `init` as well. I think they are better understood when separate instead of using one monolithic function for both. You can just put these tests in the same block where we have other tests like `head` and `last`, it will keep the test code consistent for all tests.